### DISCUSSION TITLE
Passing livewire variabels to localstorage JS using Alpine

### DISCUSSION THREAD

**Discussion by agustinus-robert** [livewire/livewire#3586] [1. Help]
Hei, sorry my bad english...

I want to send my data array from livewire component

$myarr = [];

to alpine and then i want to save that array to localstorage

i was looking sharing between data from livewire to alpine, using @entangle

Can i save data was shared @entangle to localstorage?

**Comment by ariaieboy** [comment]
yes you can do that and we have a plugin for it in alpine too:

[alpine persist plugin](https://alpinejs.dev/plugins/persist)

**Reply by agustinus-robert** [reply]
Hai, i was using this plugins and

this livewire/localstore.php

`<?php

namespace App\Http\Livewire\Memory;

use Livewire\Component;

class LocalStore extends Component
{
    public $increment;
    
    public function add(){
        $this->increment++;
    }
    
    public function render()
    {
        return view('livewire.memory.local-store');
    }
}
`

and than the view

![Screen Shot 2021-08-26 at 15 45 39](https://user-images.githubusercontent.com/61762097/130931870-b8fea352-fea2-4964-adc2-525af1e5dee8.png)


![perist](https://user-images.githubusercontent.com/61762097/130931249-bf2c413e-b2e5-42a5-8343-c2085ddb8e26.gif)

the local store is work, but if i refresh page, the localStorage gone... from livewire variable


**_But if i not using livewire_**

![Screen Shot 2021-08-26 at 15 51 24](https://user-images.githubusercontent.com/61762097/130932707-d664c617-b47b-4dd5-ab77-ba8bc2558373.png)

it can work

![notlif](https://user-images.githubusercontent.com/61762097/130932783-c74a9a51-0fa6-49d7-a9db-7b9137465581.gif)

Can you have suggestions using livewire, but the localstorage saved..

**Reply by ariaieboy** [reply]
you dont need that add function in livewire try this:

```html
<div x-data="{count: $persist(@entangle('increment'))}">
    <button @click="count++">add</button>
    <span x-text="count"></span>
</div>
```

with this code you can share your alpine state with livewire.

both livewire and alpine have good docs just read the document first.
[livewire alpine support doc](https://laravel-livewire.com/docs/2.x/alpine-js)
